modules/nlu/src/backend/election/simple-election.ts

```diff
@@ -0,0 +1,99 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+
+const NONE_INTENT = 'none'
+
+function detectAmbiguity(intents: sdk.NLU.Intent[]): boolean {
+  // +- 10% away from perfect median leads to ambiguity
+  const preds = intents
+  const perfectConfusion = 1 / preds.length
+  const low = perfectConfusion - 0.1
+  const up = perfectConfusion + 0.1
+  const confidenceVec = preds.map(p => p.confidence)
+
+  const allInRange = (vec: number[], lower: number, upper: number) =>
+    vec.map(v => _.inRange(v, lower, upper)).every(_.identity)
+
+  const ambiguous =
+    preds.length > 1 &&
+    (allInRange(confidenceVec, low, up) ||
+      (preds[0].name === NONE_INTENT && allInRange(confidenceVec.slice(1), low, up)))
+
+  return ambiguous
+}
+
+/**
+ * @description confidence of none intent becomes Math.max(none, oos)
+ * @param oos confidence of OOS
+ */
+const replaceNone = (oos: number) => (i: sdk.NLU.Intent): sdk.NLU.Intent => {
```

Comment:
This step should also be done inside nlu-core, but by doing so, we would the legacy-election behavior...

There's actually no reason why any client of `Engine` needs to be aware of our oos implementation details. We should pick either oos or none intent, but never present both...